dynamixel_controllers/src/dynamixel_controllers/joint_state_publisher.py

```diff
@@ -0,0 +1,116 @@
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+from threading import Thread
+
+import rospy
+
+from sensor_msgs.msg import JointState
+
+class JointStatePublisher():
+    def __init__(self, controller_namespace, controllers):
+        self.update_rate = 1000
+        self.state_update_rate = 50
+        self.trajectory = []
+        
+        self.controller_namespace = controller_namespace
+        self.joint_names = [c.joint_name for c in controllers]
+
+        self.joint_to_controller = {}
+        for c in controllers:
+             self.joint_to_controller[c.joint_name] = c
+            
+        self.port_to_joints = {}
+        for c in controllers:
+            if c.port_namespace not in self.port_to_joints: self.port_to_joints[c.port_namespace] = []
+            self.port_to_joints[c.port_namespace].append(c.joint_name)
+            
+        self.port_to_io = {}
+        for c in controllers:
+            if c.port_namespace in self.port_to_io: continue
+            self.port_to_io[c.port_namespace] = c.dxl_io
+
+    def initialize(self):
+        self.msg = JointState()
+        return True
+
+
+    def start(self):
+        self.running = True
+        self.state_pub = rospy.Publisher('joint_states', JointState , queue_size=100)
+
+        Thread(target=self.update_state).start()
+
+
+    def stop(self):
+        self.running = False
+
+    def update_state(self):
+        rate = rospy.Rate(self.state_update_rate)
+        while self.running and not rospy.is_shutdown():
+            self.msg.header.stamp = rospy.Time.now()
+            self.msg.name = []
+            self.msg.position = []
+            self.msg.velocity = []
+            self.msg.effort = []
+            
+            for port, joints in self.port_to_joints.items():
+                vals = []
+                for joint in joints:
+                    j = self.joint_names.index(joint)
+                    
+                    motor_id = self.joint_to_controller[joint].motor_id
+                    co = self.joint_to_controller[joint]
+                    io = self.port_to_io[port]
+
+                    self.msg.name.append(joint)
+                    po = ve = ef = 0
+                    try:
+                        ret = io.get_feedback(motor_id)
+                        po = self.raw_to_rad_pos(ret['position'],co)
+                        ve = self.raw_to_rad_spd(ret['speed'],co)
+                        ef = self.raw_to_rad_spd(ret['load'],co)
+                    except Exception as e:
+                        rospy.logerr(e)
+                    self.msg.position.append(po)
+                    self.msg.velocity.append(ve)
+                    self.msg.effort.append(ef)
+            
+            self.state_pub.publish(self.msg)
+            rate.sleep()
+
+    def raw_to_rad_pos(self, raw, c):
```

Comment:
this duplicates some of the logic already contained in the actual underlying controllers.

it might be better to subscribe to the `/.../state` topics of the underlying controllers and republish that data in the `sensor_msgs/JointState` format

as a quick hack, I've added `/joint_state` as a 2nd published topic by the meta controller, which publishes `control_msgs/FollowJointTrajectoryFeedback` whole `actual` field contains exactly all that's needed by `sensor_msgs/JointState`. I'll send out a PR with that in case anyone else is interested.